File: app/src/main/java/com/example/cb/test/mvp/presenter/MvpFragmentPresenter.java
package com.example.cb.test.mvp.presenter;

import com.example.cb.test.mvp.BasePresenter;
import com.example.cb.test.mvp.view.IMvpFragmentView;
import com.example.cb.test.rx.http.MovieHttpRequest;
import com.example.cb.test.rx.http.NewsResp;
import com.example.cb.test.rx.http.XHttpCallback;

import java.util.HashMap;
import java.util.Map;

/**
 * Created by bin on 2018/12/24.
 */
public class MvpFragmentPresenter extends BasePresenter<IMvpFragmentView> {

    public MvpFragmentPresenter(IMvpFragmentView view) {
        super(view);
    }

    public void fetchData() {
        Map<String, String> map = new HashMap<>();
        map.put("type", "top");
        map.put("key", "f323c09a114635eb935ed8dd19f7284e");
        MovieHttpRequest.getInstance().sendNewsRequest(map, new XHttpCallback<NewsResp>() {
            @Override
            public void onSuccess(NewsResp newsResp) {
                mView.onNewsSuccess(newsResp);
            }

            @Override
            public void onError(String error) {
                mView.onNewsError(error);
            }
        });
    }
}

File: app/src/main/java/com/example/cb/test/mvp/presenter/MvpFragmentPresenter.kt
package com.example.cb.test.mvp.presenter

import com.example.cb.test.base.BasePresenter
import com.example.cb.test.mvp.view.IMvpFragmentView
import com.example.cb.test.rx.http.MovieHttpRequest
import com.example.cb.test.rx.http.NewsResp
import com.example.cb.test.rx.http.XHttpCallback

class MvpFragmentPresenter(view: IMvpFragmentView) : BasePresenter<IMvpFragmentView>(view) {

    fun fetchData() {
        val map = HashMap<String, String>()
        map["type"] = "top"
        map["key"] = "f323c09a114635eb935ed8dd19f7284e"
        MovieHttpRequest.getInstance().sendNewsRequest(map, object : XHttpCallback<NewsResp> {
            override fun onSuccess(newBean: NewsResp?) {
                if (newBean != null) {
                    mView!!.onNewsSuccess(newBean)
                }
            }

            override fun onError(error: String?) {
                if (error != null) {
                    mView!!.onNewsError(error)
                }
            }
        })
    }
}